services/server/cg_server/error.py

```diff
@@ -12,17 +12,19 @@
 
 from flask import make_response
 
+
 # Print to stderr (for Google Cloud Run error tracking)
 # https://stackoverflow.com/a/14981125/4343866
 def eprint(*args, **kwargs):
     print(*args, file=sys.stderr, **kwargs)
 
 
-def handle_db_errors(conn):
+def handle_db_errors(conn, options, conn_pool):
     def decorator_db_error(func):
         @functools.wraps(func)
         def wrapper_db_error(*args, **kwargs):
             try:
+                conn = conn_pool.getconn()
                 res = func(*args, **kwargs)
             # Catch any database/SQL errors
             except psycopg2.Error as e:
```

Comment:
In the routes, you're getting a connection with `get_conn_from_pool()` and then passing it into the downstream API function. But in this function wrapper, you're pulling a different connection and then passing it to the API function, where it isn't used. I think the end result is that you're using two connections for one request. The commit/rollback protections in this error wrapper won't function since they're operating on a different connection from that in the actual API function

Instead of this:

```
@app.route("/init")
@cross_origin(origins=cors_domains)
@handle_db_errors(conn=get_conn_from_pool(connection_options, conn_pool),
                  options=connection_options,
                  conn_pool=conn_pool)
def init():
    conn = get_conn_from_pool(connection_options, conn_pool)
    return query_initial(conn)
```

We should aim for something like this:

```
@app.route("/init")
@cross_origin(origins=cors_domains)
@handle_db_errors(options=connection_options, conn_pool=conn_pool)
def init(conn): // conn is passed into this function by the @handle_db_errors decorator
    return query_initial(conn)
```